DOCS/git-usage.md

```diff
@@ -0,0 +1,104 @@
+  * If you type `git remote -v` again, you should see both "origin" and "upstream" listed now (with separate "fetch" and "push" entries for each; don't worry about that).
+
+After all of this is done, you should be all set to start adding new licenses.
+
+## Adding a new license
+
+For the following steps, we'll assume that you're adding a new license called the XYZ license.
+
+1. In the terminal, make sure you're starting on the master branch, and that it's up to date with the upstream's master:
+  * **Change to the master branch**: `git checkout master`
+  * **Pull in any updates from upstream**: `git pull upstream master`
+2. Now that it's synced with upstream, you'll create a branch into which you'll commit your changes. I'm calling it "xyz" here for the new license, but you can name it anything unique.
+  * **Create new branch**: `git checkout -b xyz`
+3. When the new branch is created, git will also automatically change it so that the new branch is your working branch. You can type `git status` to see your current working branch: ![screenshot from typing `git status` on clean repo](/DOCS/images/git-usage-git-status-clean.png)
+4. Now, since you're on the new branch, go ahead and create the new files in your text editor for the license you're adding. The two files for the "XYZ" license would be:
+  * license template XML file: `src/XYZ.xml`
+  * test text file: `test/simpleTestForGenerator/XML.txt`
+5. After the files are created and saved, the files exist on your hard drive, but they are not yet saved to the new branch. If you type `git status` again, git sees the files are there but are not currently tracked by Git: ![screenshot from typing `git status` with untracked files](/DOCS/images/git-usage-git-status-untracked.png)
+6. The next step is to "**add**" the files, so that Git will start tracking them. The easiest way to do this is to specify "src" and "test" in the add command, which would add any new files that are found in the "src/" or "test/" directories:
+  * **Add the files to be staged**: `git add src test`
+7. Now the files are "staged", but confusingly they haven't yet been committed to the branch. If you type `git status` again it will tell you this: ~[screenshot from typing `git status` with added files](/DOCS/images/git-usage-git-status-added.png)
+8. Next, you'll "commit" the changes, which actually inserts them into your local branch:
+  * **Commit the files**: `git commit -s -m "Add XYZ license"`
+  * ![screenshot from typing `git commit`](/DOCS/images/git-usage-git-commit.png)
+  * In the commit command, the `-m` flag lets you add a one-line message to the commit. If you omit `-m` and the quoted message, it will take you to a text editor where you can type a longer message.
+  * The `-s` flag adds the DCO `Signed-off-by:` line to the commit message. If you're having issues when using this flag, let me know, there might be one other piece of setup needed for you to enable it.
```

Comment:
With all respect to the DCO and it’s usability, do we really _need_ it here?

Does editing through the GitHub webUI actually Sign Off? If not, I’d keep the DCO as optional, not to overburden contributors.

Reply:
We haven't really been using the DCO for this repository - I know @swinslow and @goneall have `Signed-off-by` lines in their commits here, but I imagine it isn't required for this repository, right?

Reply:
Personally, I always sign-off my commits independent of any requirements.  As to requiring others to sign-off, I believe it is quite important for any code contributions (e.g. a fix to any of the scripts or utilities used in the license-list-XML repo), but not as important for license text submittals.  I'll leave it to others to decide if we want to require sign-offs for license related submittals.

Reply:
I agree it’s a good thing™ and I really shout set my git config to do that automatically for me.

But if we want to keep these instructions as simple as possible for the less-technically-experienced, unless it is a hard requirement, I would leave it out. Especially if editing via GitHub’s WebUI does not sign off.

Perhaps move it lower as a suggestion instead for those who (want to) know how to do that.

Reply:
All good points, thanks for this.

Given that the license-list-XML repo doesn't currently require DCO sign-offs, I'm inclined to agree that the base instructions should not describe it as a required part of the flow. I'm removing it, and adding separate details to the bottom of the document in case folks choose to include it.